### DISCUSSION TITLE
Dependency issue when integrate API into `@client/pages`

### DISCUSSION THREAD

**Discussion by VietAnh1010** [joulev/ezkomment#93] [Questions]
Currently, `@server/pages/api` cannot be moved into `@client/pages/api` as the API require some dependencies that are not listed in `@server/package.json`. Adding dependecies into `@client` is  not a good idea.

How should we resolve this dependency problem?

**Comment by joulev** [comment]
Running `npm install` will also "install" `@ezkomment/client` and `@ezkomment/server`, so from `packages/client` you can call server lib like normal NPM dependencies. See [NPM workspaces documentation](https://docs.npmjs.com/cli/v8/using-npm/workspaces) and 1982bd831988c8dbc2a05ca199e24d23d9135f26. (please revert the commit though)

However, if you have made your mind up and decided to ditch Express in favour of Next.js API routes, it's better to move all your lib code to `packages/client`. For example, `lib/client` for client-side lib code and `lib/server` for server-side lib code. Do note that Express's `Request` and `Response` objects are _different_ from Next.js' (e.g. Next.js doesn't have `res.cookie()` and `res.clearCookie()`, and the migration process from an Express-based codebase to a Next.js codebase is not too trivial.

In the case of moving to Next.js entirely, I will convert this from a monorepo to a normal repository and clear all "client" status of the Next.js app.

I do not recommend maintaining a Next.js-based backend and an Express backend at the same time. It just doesn't make sense, let's only choose one of them. Both are fine to me.

**Comment by joulev** [comment]
Apparently NPM workspaces don't work on Vercel, I will change to `yarn`. If you see this, install `yarn`.

**Comment by VietAnh1010** [comment]
Installed `yarn`.

> I do not recommend maintaining a Next.js-based backend and an Express backend at the same time. It just doesn't make sense, let's only choose one of them. Both are fine to me.

There is no need for a full-scale Express App. Let's choose Next.js. Also, using Next.js will simplify the deployment process. 

A converter will be used to temporarily utilize the Express-based code before the migration is completed.

**Comment by joulev** [comment]
Working example: https://ezkomment-diymdjav7-joulev.vercel.app/api/hello

However, since Next.js will be used instead of Express, I suggest migrating as soon as possible.

**Comment by VietAnh1010** [comment]
Got it, will start migrating tomorrow.
